internal/controller/controllers/common.go

```diff
@@ -41,18 +41,19 @@ func getPullSecret(ctx context.Context, c client.Client, name, namespace string)
 	return string(data), nil
 }
 
-func getInfraEnvByClusterDeployment(ctx context.Context, c client.Client, clusterDeployment *hivev1.ClusterDeployment) (*aiv1beta1.InfraEnv, error) {
+func getInfraEnvByClusterDeployment(ctx context.Context, c client.Client, name, namespace string) (*aiv1beta1.InfraEnv, error) {
 	infraEnvs := &aiv1beta1.InfraEnvList{}
 	if err := c.List(ctx, infraEnvs); err != nil {
-		logrus.WithError(err).Errorf("failed to search for infraEnv for clusterDeployment %s", clusterDeployment.Name)
+		logrus.WithError(err).Errorf("failed to search for infraEnv for clusterDeployment %s", name)
```

Comment:
Suggesting to get the log from the context, this way you will get the request ID and it will be easier to see the full flow when debugging, or pass a log as another parameter.

Reply:
Sure, added.